### PR TITLE
fix(matterbridge): fix Matterbridge setup

### PR DESCRIPTION
**Pull request by Antreesy**
### ☑️ Resolves

* Fix Matterbridge issues with setup
  * Don't save selected result, as it was resetted anyway
  * Move hardcoded types to separate file
  * Minor code refactor
  * Skip TLS for local NC server, optionally skip TLS for target NC server (self-signed certificate)

>[!TIP]
> To test:
> clone app https://github.com/nextcloud/talk_matterbridge to `apps-*` directory on your server
> make binaries (see app README)
> In Administration settings -> Apps -> Enable Talk Matterbridge
> In Administration settings -> Talk -> Enable Matterbridge integration
> In Conversation settings -> Matterbridge -> Select Nextcloud Talk:
> - URL: any host works (tested with https://nextcloud.local to itself and to https://stable29.local)
> - user: userid of user, who will be posting messages on behalf of Matterbridge (should be added to this room)
> - conversation: room token

## 🖌️ UI Checklist

Before | After
-- | --
image | image
image | image

### 🚧 Tasks

- [ ] Test without TLS check (need proper certificates on a receiving end)

### 🏁 Checklist

- [ ] 🌏 Tested with different browsers / clients:
  - [x] Chromium (Chrome / Edge / Opera / Brave)
  - [ ] Firefox
  - [ ] Safari
  - [ ] Talk Desktop
  - [ ] Not risky to browser differences / client

### 🚧 Tasks

- [ ] ...

### 🏁 Checklist

- [ ] ⛑️ Tests (unit and/or integration) are included or not possible
- [ ] 📘 API documentation in `docs/` has been updated or is not required
- [ ] 🔖 Capability is added or not needed

### CODE REVIEW

**Comment by nickvergessen** [comment]
### More info icon is wrong color
Bildschirmfoto vom 2024-11-15 11-13-55

### Section icon is misaligned
Bildschirmfoto vom 2024-11-15 11-14-01

File: src/components/ConversationSettings/Matterbridge/matterbridgeTypes.ts [hunk 1]
```diff
@@ -0,0 +1,386 @@
...
+				icon: 'icon-user',
+			},
+			password: {
+				type: 'password',
+				placeholder: t('spreed', 'User password'),
+				icon: 'icon-category-auth',
+			},
+			channel: {
+				type: 'text',
+				placeholder: t('spreed', 'Talk conversation'),
```
Comment (DorraJaouad) [inline src/components/ConversationSettings/Matterbridge/matterbridgeTypes.ts:48]: Change to `Talk conversation token` ?

**Review by DorraJaouad — APPROVED** [review]
Code-wise looks fine yet, I couldn't test due to type mismatch error.

File: src/components/ConversationSettings/Matterbridge/matterbridgeTypes.ts (hunk 1 above)
Comment (Antreesy) [reply src/components/ConversationSettings/Matterbridge/matterbridgeTypes.ts:48]: Follow-up, BridgePart needs migration to Vue components, and it's unrelated to fix TBH

**Pull request merged**